### PR TITLE
stmhal: modified uart any function to return bytes

### PR DESCRIPTION
**Pull request by ryannathans**
My first contribution to the project. I've modified uart any to return number of bytes in the buffer rather than true/false. Tested and works well in my project. Addresses desired functionality discussed in #1533

### CODE REVIEW

**Comment by twitchyliquid64** [comment]
+1 can we mainline this plz need in my project

File: stmhal/uart.c [hunk 1]
```diff
@@ -282,9 +282,17 @@ bool uart_init(pyb_uart_obj_t *uart_obj, uint32_t baudrate) {
...
 */
 
-bool uart_rx_any(pyb_uart_obj_t *self) {
-    return self->read_buf_tail != self->read_buf_head
-        || __HAL_UART_GET_FLAG(&self->uart, UART_FLAG_RXNE) != RESET;
+mp_uint_t uart_rx_any(pyb_uart_obj_t *self) {
+    int buffer_bytes = self->read_buf_head - self->read_buf_tail;
+    if (buffer_bytes < 0) {
+        return buffer_bytes + self->read_buf_len;
+    }
```
Comment (dpgeorge) [inline stmhal/uart.c]: Code style: else should be on same line as }

**Comment by ryannathans** [comment]
@dpgeorge fixed

**Comment by dpgeorge** [comment]
Thanks!

Merged in f99491cbf743d51fd2aa7332c0e03da64448d60a with slightly reworded commit message, and modified tests/pyb/uart.py.exp.

**Pull request closed without merging**